paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -853,6 +854,16 @@ class PoolingMKLDNNHandler : public MKLDNNHandlerT<T, mkldnn::pooling_forward,
         CorrectOutputSize(src_tz, dst_tz, ksize, paddings, strides,
                           mkldnn_paddings[1]);
       }
+
+      if (adaptive) {
+        ksize[0] = static_cast<int>(
+            ceil(static_cast<double>(src_tz[src_tz.size() - 2] / ksize[0])));
+        ksize[1] = static_cast<int>(
+            ceil(static_cast<double>(src_tz[src_tz.size() - 1] / ksize[1])));
```

Comment:
I guess you want to round up the kernel size, thus you need to cast to floating point at least one arg of division operator, otherwise both sides are integers and we have integer division. If integer division is that what you want, then cast to double is not necessary.
```suggestion
        ksize[0] = static_cast<int>(
            ceil(static_cast<double>(src_tz[src_tz.size() - 2]) / ksize[0]));
        ksize[1] = static_cast<int>(
            ceil(static_cast<double>(src_tz[src_tz.size() - 1]) / ksize[1]));
```

Reply:
I'd suggest to have UT which distinguish those two slightly different rounding cases.

Reply:
@arogowie-intel After some analysis of reference solution I restricted supported cases only to those when src_tz dims are divisible by corressponding ksize dims. This is because when we have some rounding applied then we may end up that during one pooling operation (multiple poolings over same input) pooling window size may vary , which is not supported by oneDNN.